### DISCUSSION TITLE
Weblate 4.3.x bug?

### DISCUSSION THREAD

**Discussion by mareksuscak** [WeblateOrg/weblate#4805] [Q&A]
Ever since we upgraded to Weblate 4.3, we've been getting the "too many open files" error which we have never seen before and we've been using Weblate for years. I ran `git gc` a number of times and we have not added new repositories or translations in a long while so the environment is the same. It seems that 4.3 introduced some change and possibly a leak that's causing it. 

There's also one more issue we're running into where Weblate keeps removing our non-primary translations from one of the projects and  the only change we did there is that we changed the file pattern and the file pattern includes a `+` character.

Something along these lines:

```
translations/messages+intl-icu.*.yaml
```

I wonder if this has to do anything with the "too many open files" issue and also why weblate keeps removing those translations. For more context, when I click Add Translation and select the missing languages, they're added no problem and also the strings from those files are added successfully to weblate database (though I don't know if weblate displays the old entries from before the renaming).

I'd appreciate any help.

**Comment by nijel** [comment]
The issue with + in mask will be addressed in 4.3.2 and is fixed by 155b489c83a323ea5b54de2a37e77787c343c998.

Too many open files is unrelated and I haven't seen it so far, can check which files have the process open?

**Reply by mareksuscak** [reply]
We're running the official Docker container on Kubernetes (GKE) and `lsof` is not available in the slim version of the Debian docker image. I guess I need to build a custom docker image w/ `lsof` utility or is there any other way in your knowledge?

**Reply by nijel** [reply]
You can install anything during the container runtime: `apt update && apt install lsof`

You can also list the process open files in proc filesystem by looking at `/proc/<PID>/fd` (but that won't give much information on sockets).

**Reply by mareksuscak** [reply]
I can't because the container is not running as root and `kubectl exec` [does not support switching users](https://github.com/kubernetes/kubernetes/issues/30656) yet.

![image](https://user-images.githubusercontent.com/613647/98114765-09a2ca00-1ea6-11eb-9c72-8e6c4b85d06c.png)

**Reply by nijel** [reply]
Ah okay, I thought it works same as with pure docker. Anyway the proc filesystem should still work and would give an answer in case these are real files and not sockets...
